### PR TITLE
Adjusting TESTING.md for a single test run

### LINKED ISSUE #14014: [Testing Doc Update] Verify and update the test documentation in Core if "Test case filtering" instruction is outdated
### Describe the bug

Based on the [Operations & Project Health Meeting held on 2024-05-30](https://forum.opensearch.org/t/operations-project-health-meeting-2024-05-30/19633), a community member raised that the **Specific Test** execution documentation covered in the [Test Case Filtering](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#test-case-filtering) section of [Testing.md](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md) needs to be updated, as it does not work for them.

We need to verify if the documentation for running individual tests requires an update.

### Related component

Other

### To Reproduce

Run a single test case (variants)

```
./gradlew test -Dtests.class=org.opensearch.package.ClassName
./gradlew test "-Dtests.class=*.ClassName"
```

### Expected behavior

It should successfully execute single test

### Additional Details

_No response_

### PR DESCRIPTION
**Pull request by DaniilRoman**
### Description
Adjusting the gradle commands for running a single test and running the filtered tests.
According a Gradle [docs](https://docs.gradle.org/current/userguide/intro_multi_project_builds.html#sec:multiproject_build_and_test) for multi modules projects.

Below the screenshots of how it worked on my machine: 
1. Without specifying a module name ❌ 
Screenshot 2024-06-18 at 23 07 21
2. With specifying a module name ✅ 
Screenshot 2024-06-18 at 23 07 54

### Related Issues
Resolves #14014

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -91,18 +91,22 @@ This will instruct all JVMs (including any that run cli tools such as creating t
 -   `tests.class` is a class-filtering shell-like glob pattern
```
Comment (reta) [inline TESTING.md:91]: I think this test case filtering patterns are not applicable anymore, right?

**Review by reta — APPROVED** [review]

File: TESTING.md (hunk 1 above)
Comment (DaniilRoman) [reply TESTING.md]: You're right, thank you for the comment!
I deleted it and substituted with an example

File: TESTING.md [hunk 2]
```diff
@@ -88,25 +88,27 @@ This will instruct all JVMs (including any that run cli tools such as creating t
...
 
 Run any test methods that contain *esi* (e.g.: .r*esi*ze.)
 
-    ./gradlew test "-Dtests.method=*esi*"
+    ./gradlew module:test --tests "*esi*"
 
 Run all tests that are waiting for a bugfix (disabled by default)
 
-    ./gradlew test -Dtests.filter=@awaitsfix
+    ./gradlew module:test --tests "*@awaitsfix"
```
Comment (reta) [inline TESTING.md:111]:
```suggestion
./gradlew test -Dtests.filter=@awaitsfix
```

   So the filtering by annotation does not work, using `-Dtests.filter` is legit

**Review by reta — CHANGES_REQUESTED** [review]

File: TESTING.md (hunk 2 above)
Comment (DaniilRoman) [reply TESTING.md:111]: Ahh, my bad, thank you for adding a suggestion change 🙏

**Review by reta — APPROVED** [review]

**Pull request merged**